SimpleMathJaxHooks.php

```diff
@@ -31,13 +31,11 @@ public static function renderChem($tex, array $args, Parser $parser, PPFrame $fr
 	}
 
 	private static function renderTex($tex, $parser) {
-		$parser->getOutput()->addModules( 'ext.SimpleMathJax' ); 
+		$parser->getOutput()->addModules( 'ext.SimpleMathJax' );
 		$parser->getOutput()->addModules( 'ext.SimpleMathJax.mobile' ); // For MobileFrontend
-		return ["<span style='opacity:.5'>[math]${tex}[/math]</span>", 'markerType'=>'nowiki'];
+		$attributes = [ "style" => "opacity:.5" ];
+		Hooks::run( "SimpleMathJaxAttr", [ &$attributes, $tex ] );
```

Comment:
How about changing it like this? Most users won't need the hook. So the default value of $wgSmjEnableAttributesHook will be false.
```suggestion
		if( $wgSmjEnableAttributesHook ) Hooks::run( "SimpleMathJaxAttributes", [ &$attributes, $tex ] );
```

Reply:
hook & setting variable name candidates:
Hook | Setting variable
------ | ------------------
SimpleMathJaxAttributes | wgSmjEnableAttributesHook
SmjAttributes | wgSmjEnableAttributesHook 
SimpleMathJaxOverrideAttributes | wgSmjEnableOverrideAttributesHook
SmjOverrideAttributes | wgSmjEnableOverrideAttributesHook

Reply:
### Re: Naming.

I like `SimpleMathJaxAttributes` -- keeping it readable.  Maybe even `SimpleMathJaxSpanAttributes`

### Re: adding a variable.

Why not just add the hook?  What would be the purpose of adding the variable?  None of the other examples you point to have a setting to trigger the invocation of their hooks.

If you just add the hook and no other extensions try to use it, then the current behavior does not change.

If someone added the hook, they would want it used, right?  Under what case would a hook be added but it would be desirable not to use it?

For example, again, the use case I'm looking at would be adding the attributes so that the Lingo extension ignores math formulas.  If you accept the hook, I would then create a PR against Lingo so that it had a hooked function for `SimpleMathJaxAttributes`.

This would mean that:
* If Lingo is installed, but the SimpleMathJax is not installed, nothing changes.
* If SimpleMathJax is installed but Lingo is not installed, nothing changes.
* If both extensions are installed and an end user happens to add a formula with a word that Lingo recognizes, math is rendered correctly.

If we add the configuration variable as well as the hook, then the end user has to figure out why his formula isn't rendering and then ask the wiki administrator to set a configuration variable.

Reply:
Thanks for your feedback. Let's just add the hook 'SimpleMathJaxAttributes'.
```
Hooks::run( "SimpleMathJaxAttributes", [ &$attributes, $tex ] );
```